### PR TITLE
Give class template context to conditional checker

### PR DESCRIPTION
**Pull request by KmeCnin**
This PR fixes bug reported here: https://github.com/phpstan/phpstan/issues/8609

### CODE REVIEW

File: tests/PHPStan/Generics/data/bug8609.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types = 1);
+
+namespace Generics\Bug8609;
```
Comment (staabm) [inline tests/PHPStan/Generics/data/bug8609.php]: it seems you are adding a new test-case here, but you did not analyze this file from any of the *Test classes.
you need to make sure that this file is analyzed from the correct *Test class - the one which is testing the rule which emits the error you are trying to fix

File: src/Rules/PhpDoc/MethodConditionalReturnTypeRule.php [hunk 2]
```diff
@@ -31,7 +31,12 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		return $this->helper->check($variants[0]);
+		$variant = $variants[0];
+		$templateTypeMap = $node->getClassReflection()->getTemplateTypeMap()->union(
```
Comment (ondrejmirtes) [inline src/Rules/PhpDoc/MethodConditionalReturnTypeRule.php]: `union` isn't the best operation for this. I suggest `array_merge` instead, like here: https://github.com/phpstan/phpstan-src/blob/b250a8bf799b6d22b979260ecfa7539f16449a0f/src/Analyser/NameScope.php#L164-L167

File: tests/PHPStan/Generics/data/bug8609.php (hunk 1 above)
Comment (ondrejmirtes) [reply tests/PHPStan/Generics/data/bug8609.php]: 1) File name needs to be in `bug-8609.php` format.
2) File needs to be analysed by tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Additionally, we need to test that this actually works. A new file tested in NodeScopeResolverTest would be nice.

**Comment by ondrejmirtes** [comment]
Finished it myself :) Thank you!

**Review by ondrejmirtes — APPROVED** [review]

**Pull request merged**